### PR TITLE
Returning null in template should nullify parent template

### LINKED ISSUE #5148: null template should override inherited template
Currently, if an element extends a parent element that _has_ a template, there's no simple way to override that template and get an element with no template or shadow root.

```js
class ParentElement {
  static get template() { html`Haha! You're stuck with my template!`; }
}

class ChildElement extends ParentElement {
  // I don't want your template! Or anyone's template! Leave me alone!
  static get template() { return null; }
}
```

@kevinpschaaf suggested that an explicit falsey value like `null` should override the parent template. If the child template getter returns `undefined`, it will continue to inherit the parent template.

### PR DESCRIPTION
**Pull request by TimvdLippe**
The suspicion of @kevinpschaaf in https://github.com/Polymer/polymer/issues/5148#issuecomment-372491826 was correct, in that moving the assignment fixed the issue.
### Reference Issue
Fixes #5148

### CODE REVIEW

File: test/unit/inheritance.html [hunk 1]
```diff
@@ -237,6 +255,23 @@
...
+      el = document.createElement('child-with-no-template');
+      document.body.appendChild(el);
+    });
+
+    teardown(function() {
+      document.body.removeChild(el);
+    });
+
+    test('returning null nullifies the parent template', function() {
+      assert.isNotOk(el.shadowRoot);
```
Comment (azakus) [inline test/unit/inheritance.html]: Let's make this `assert.equal(el.shadowRoot, null)`

File: lib/mixins/element-mixin.html [hunk 2]
```diff
@@ -311,9 +311,9 @@
           } else {
             template = template.cloneNode(true);
           }
-          this.prototype._template = template;
         }
 
+        this.prototype._template = template;
```
Comment (azakus) [inline lib/mixins/element-mixin.html]: Nice easy fix!

**Comment by arthurevans** [comment]
So, follow up: With this fix, does any falsy value work, as originally documented, or does it have to be `null`?

**Comment by TimvdLippe** [comment]
Any falsy value should work.

**Comment by arthurevans** [comment]
Thanks Tim!

**Pull request merged**